vault/namespace_store.go

```diff
@@ -817,6 +817,9 @@ func (ns *NamespaceStore) ResolveNamespaceFromRequest(nsHeader, reqPath string)
 // a locked namespace, starting from one of the root children,
 // ending at the namespace provided as a argument to the function.
 func (ns *NamespaceStore) GetLockingNamespace(n *namespace.Namespace) *namespace.Namespace {
```

Comment:
Should this just acquire the lock in the exported version and the other (exported) methods call `getLockingNamespace(...)` which does all the rest of the work, to be called from `LockNamespace(...)`/`UnlockNamespace(...)` since two unlocks could race and we should fail one of them? Seems like we'd not want want to grab a read lock only to release and re-acquire a write lock.